compiler/rustc_middle/src/ty/context.rs

```diff
@@ -406,6 +408,39 @@ impl<'tcx> Deref for TyCtxt<'tcx> {
     }
 }
 
+// `TyCtxt` may contain non-shared data structures only at initialization
+// time. We can determine the thread safety of `Tyctxt` by checking at
+// runtime whether these non-shared data structures actually exist.
+unsafe impl<'tcx> DynSendSyncCheck for TyCtxt<'tcx> {
+    #[inline]
+    fn check_send_sync(&self) {
```

Comment:
Can you use `let GlobalCtxt { a, b, c } = self` for exhaustiveness checking?

Reply:
Yea, it makes sense!